kotlin/animate-images-with-image-overlay/src/main/java/com/esri/arcgisruntime/sample/animateimageswithimageoverlay/MainActivity.kt

```diff
@@ -0,0 +1,237 @@
+import com.esri.arcgisruntime.mapping.view.ImageOverlay
+import kotlinx.android.synthetic.main.activity_main.*
+import java.io.File
+import java.util.Arrays
+import java.util.Timer
+import kotlin.concurrent.fixedRateTimer
+
+class MainActivity : AppCompatActivity() {
+
+  private val imageFrames: MutableList<ImageFrame> by lazy { mutableListOf<ImageFrame>() }
+  private var imageIndex: Int = 0
+
+  private var timer: Timer? = null
+  private var isTimerRunning = true
+  private var period: Long = 67
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    // create a new tiled layer from the World Dark Gray Base REST service
+    val worldDarkGrayBasemap =
+      ArcGISTiledLayer("https://services.arcgisonline.com/arcgis/rest/services/Canvas/World_Dark_Gray_Base/MapServer")
+
+    // create a new elevation source from the Terrain3D REST service
+    val elevationSource =
+      ArcGISTiledElevationSource("https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer")
+
+    // create an envelope of the pacific southwest sector for displaying the image frame
+    val pointForImageFrame =
+      Point(-120.0724273439448, 35.131016955536694, SpatialReferences.getWgs84())
+    val pacificSouthwestEnvelope = Envelope(pointForImageFrame, 15.09589635986124, -14.3770441522488)
+
+    // create a camera, looking at the pacific southwest sector
+    val observationPoint = Point(-116.621, 24.7773, 856977.0)
+    val camera = Camera(observationPoint, 353.994, 48.5495, 0.0)
+    val pacificSouthwestViewpoint = Viewpoint(pacificSouthwestEnvelope, camera)
+
+    // create a scene with the dark gray basemap and elevation source
+    val darkGrayScene =
+      ArcGISScene(Basemap(worldDarkGrayBasemap), Surface(listOf(elevationSource))).apply {
+        // set the pacific southwest as the initial viewpoint
+        initialViewpoint = pacificSouthwestViewpoint
+      }
+
+    // create the scene view
+    sceneView.apply {
+      // add the scene to the scene view
+      scene = darkGrayScene
+      // create and append an image overlay to the scene view
+      imageOverlays.add(ImageOverlay())
+      // create a touch listener
+      setOnTouchListener(object : DefaultSceneViewOnTouchListener(sceneView) {
+        // close the options sheet when the map is tapped
+        override fun onTouch(view: View?, motionEvent: MotionEvent?): Boolean {
+          if (fab.isExpanded) {
+            fab.isExpanded = false
+          }
+          return super.onTouch(view, motionEvent)
+        }
+      })
+      // ensure the floating action button moves to be above the attribution view
+      addAttributionViewLayoutChangeListener { _, _, _, _, bottom, _, _, _, oldBottom ->
+        val heightDelta = bottom - oldBottom
+        (fab.layoutParams as CoordinatorLayout.LayoutParams).bottomMargin += heightDelta
+      }
+    }
```

Comment:
Since this is all UI set up, could we abstract this into its own method to "hide" the UI logic? I've made similar suggestions before for Hudson's PRs, since it allows the focus of the sample to be more on our API, with the option for the customer to look through the code to see how the UI is handled later.